tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -2965,15 +3144,18 @@ private[optimizer] abstract class OptimizerCore(
   }
 
   private def fold3WayComparison(canBeEqual: Boolean, canBeLessThan: Boolean,
-      canBeGreaterThan: Boolean, lhs: Tree, rhs: Tree)(
-      implicit pos: Position): Tree = {
+      canBeGreaterThan: Boolean, lhs: PreTransform, rhs: PreTransform)(
+      implicit pos: Position): PreTransform = {
     import BinaryOp._
     if (canBeEqual) {
       if (canBeLessThan) {
-        if (canBeGreaterThan)
-          Block(keepOnlySideEffects(lhs), keepOnlySideEffects(rhs), BooleanLiteral(true))
-        else
+        if (canBeGreaterThan) {
+          PreTransBlock(
+              Block(finishTransformStat(lhs), finishTransformStat(rhs)),
+              PreTransLit(BooleanLiteral(true)))
```

Comment:
Consider:

```scala
PreTransBlock(
              Right(finishTransformStat(lhs)) :: Right(finishTransformStat(rhs)) :: Nil,
              PreTransLit(BooleanLiteral(true)))
```

or alternatively giving `PreTransBlock` an `apply` method that can flatten blocks. This would allow us to flatten skips earlier and this seems to be a rather common case here.